ui/src/importer/importerTaskDetails/ImporterTaskDetailsComponents/ImporterReportMode.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Label } from 'semantic-ui-react';
+
+export class ImporterReportMode extends React.Component {
+  render() {
+    const { data } = this.props;
+
+    const labels = {
+      IMPORT: (
+        <Label color="blue" basic>
+          Import
+        </Label>
+      ),
+      DELETE: (
+        <Label color="red" basic>
+          Delete
+        </Label>
+      ),
+      PREVIEW_IMPORT: (
+        <Label color="teal" basic>
+          Preview (import)
+        </Label>
+      ),
+      PREVIEW_DELETE: (
+        <Label color="teal" basic>
+          Preview (delete)
+        </Label>
+      ),
+      ERROR: (
+        <Label color="red" basic>
+          Error
+        </Label>
+      ),
+    };
+
```

Comment:
Maybe I'm completely wrong here, but I think it's bad idea to have non needed code in the render because it only can use one thread. Maybe it would be better to have the labels object in the constructor?

Reply:
I am not sure what do you mean by `render because it only can use one thread`, could you explain ? Could be in person.
However I agree with the comment, there is no reason to reinitialise the label const each time we re-render the component, since the label mapping is "static"